extmod/modssl_mbedtls.c

```diff
@@ -236,6 +318,55 @@ STATIC mp_obj_t ssl_context___del__(mp_obj_t self_in) {
 STATIC MP_DEFINE_CONST_FUN_OBJ_1(ssl_context___del___obj, ssl_context___del__);
 #endif
 
+// Ciphersuites
+
+STATIC mp_obj_t ssl_context_get_ciphers(mp_obj_t self_in) {
+    mp_obj_t list = mp_obj_new_list(0, NULL);
+    const int *cipher_list = mbedtls_ssl_list_ciphersuites();
+    while (*cipher_list) {
+        const char *cipher_name = mbedtls_ssl_get_ciphersuite_name(*cipher_list);
+        mp_obj_list_append(list,
+            MP_OBJ_FROM_PTR(mp_obj_new_str(cipher_name,
+                strlen(cipher_name))));
+        cipher_list++;
+        if (!*cipher_list) {
+            break;
+        }
+    }
+    return list;
+}
+
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(ssl_context_get_ciphers_obj, ssl_context_get_ciphers);
+
+
+STATIC mp_obj_t ssl_context_set_ciphers(mp_obj_t self_in, mp_obj_t ciphersuite) {
+    // check that ciphersuite is a list
+    if (mp_obj_is_type(ciphersuite, &mp_type_list)) {
```

Comment:
Better to use `mp_obj_get_array(ciphersuite, &len, &items)`.  That will check for tuple or list and raise an exception if not.

Reply:
`mp_obj_get_array(ciphersuite, &len, &items)`

Not sure how to implement this, i.e. do I need to declare `len` and `items` first? or pass `ciphersuite->len` etc?